crates/proto/src/lib.rs

```diff
@@ -71,10 +75,50 @@ pub use crate::xfer::secure_dns_handle::SecureDnsHandle;
 #[doc(hidden)]
 pub use crate::xfer::{BufDnsStreamHandle, BufStreamHandle};
 
+#[cfg(feature = "tokio-io")]
+#[doc(hidden)]
+pub mod iocompat {
+    use std::io;
+    use std::pin::Pin;
+    use std::task::{Context, Poll};
+
+    use futures::io::{AsyncRead, AsyncWrite};
+    use tokio::io::{AsyncRead as AsyncRead02, AsyncWrite as AsyncWrite02};
+
+    /// Conversion from `tokio::io::{AsyncRead, AsyncWrite}` to `std::io::{AsyncRead, AsyncWrite}`
+    pub struct Compat02As03<T>(pub T);
```

Comment:
Can we maybe name this something like `AsyncIo02As03`?

Reply:
Sure.